usng4j-impl/src/main/java/org/codice/usng4j/impl/UtmUpsCoordinateImpl.java

```diff
@@ -0,0 +1,292 @@
+  public double getNorthing() {
+    return northing;
+  }
+
+  @Override
+  public double getNorthingWithOffset() {
+    return isUTM()
+            && ((getNSIndicator() != null && getNSIndicator().equals(NORTH))
+                || utmNorthenBands.contains(getLattitudeBand()))
+        ? getNorthing()
+        : getNorthing() - NORTHING_OFFSET;
+  }
+
+  @Override
+  public int getZoneNumber() {
+    return zone;
+  }
+
+  @Override
+  public Character getLattitudeBand() {
+    return getLatitudeBand();
+  }
+
+  @Override
+  public Character getLatitudeBand() {
+    return latitudeBand;
+  }
+
+  @Override
+  public NSIndicator getNSIndicator() {
+    return nsIndicator == null && getLattitudeBand() != null
+        ? calculateNSIndicatorFromLatBand(getLattitudeBand())
+        : nsIndicator;
+  }
+
+  private static NSIndicator calculateNSIndicatorFromLatBand(
+      @Nonnull final Character latitudeBand) {
+    return utmNorthenBands.contains(latitudeBand) || upsNorthenBands.contains(latitudeBand)
+        ? NORTH
+        : SOUTH;
+  }
+
+  @Override
+  public CoordinatePrecision getPrecision() {
+    return precision;
+  }
+
+  @Override
+  public double getEasting() {
+    return easting;
+  }
+
+  /**
+   * @param utmUpsString a UTM/UPS formatted string. e.g. {@code 10Q 123456 0123456}
+   * @return an object representation of 'utmUpsString'
+   * @throws ParseException when 'utmUpsString' isn't correctly formatted.
+   */
+  static UtmUpsCoordinate parseUtmUpsString(final String utmUpsString) throws ParseException {
+    final Matcher matcher = utmUpsRegexp.matcher(utmUpsString);
+    if (!matcher.matches()) {
+      handleUnsuccessfulParsing(utmUpsString);
+    }
+    final String zoneNumber = matcher.group(ZONE_NUMBER_RE_GROUP);
+    final String latitudeBandString = matcher.group(LATITUDE_BAND_RE_GROUP);
+    final int easting = Integer.parseInt(matcher.group(EASTING_RE_GROUP));
+    final int northing = Integer.parseInt(matcher.group(NORTHING_RE_GROUP));
+    final String nsIndicatorString = matcher.group(NS_INDICATOR_RE_GROUP).trim();
+    final NSIndicator nsIndicator =
+        nsIndicatorString.length() > 0 ? nsIndicatorString.charAt(0) == 'N' ? NORTH : SOUTH : null;
+    return fromZoneBandNorthingEastingNSI(
+        zoneNumber.length() > 0 ? Integer.parseInt(zoneNumber) : 0,
+        latitudeBandString.length() > 0 ? latitudeBandString.charAt(0) : null,
+        easting,
+        northing,
+        nsIndicator);
+  }
+
+  private static void handleUnsuccessfulParsing(final String invalidInput) throws ParseException {
+    throw new ParseException(
+        String.format(
+            "Supplied argument '%s' is not a valid UTM/UPS formatted String.", invalidInput),
+        0);
+  }
+
+  @Override
+  public boolean isUTM() {
+    return getZoneNumber() > 0 || isUpsOverlapException(this);
+  }
+
+  static boolean isUpsOverlapException(final UtmUpsCoordinate candidateCoordinate) {
+    // TODO:  implement overlap checking
```

Comment:
❓ Would it be better to throw an `UnsupportedOperationException` for now instead of returning false?

Either way, is there a ticket to resolve this that could be referenced here?

Reply:
The fix has been posted.